app/src/androidTest/kotlin/at/bitfire/davdroid/resource/LocalCalendarTest.kt

```diff
@@ -161,19 +162,57 @@ class LocalCalendarTest {
 
     @Test
     fun testRemoveNotDirtyMarked_IdLargerThanIntMaxValue() {
-        val id = androidCalendar.addEvent(Entity(contentValuesOf(
+        val idDirty0 = androidCalendar.addEvent(Entity(contentValuesOf(
             Events.CALENDAR_ID to androidCalendar.id,
             Events._ID to Int.MAX_VALUE.toLong() + 10,
             Events.DTSTART to System.currentTimeMillis(),
             Events.DTEND to System.currentTimeMillis(),
-            Events.TITLE to "Some Event",
+            Events.TITLE to "Some Event 1",
             Events.DIRTY to 0,
             AndroidEvent2.COLUMN_FLAGS to 123
         )))
+        val idDirtyNull = androidCalendar.addEvent(Entity(contentValuesOf(
+            Events.CALENDAR_ID to androidCalendar.id,
+            Events._ID to Int.MAX_VALUE.toLong() + 11,
+            Events.DTSTART to System.currentTimeMillis(),
+            Events.DTEND to System.currentTimeMillis(),
+            Events.TITLE to "Some Event 2",
+            Events.DIRTY to null,
+            AndroidEvent2.COLUMN_FLAGS to 123
+        )))
 
         calendar.removeNotDirtyMarked(123)
 
-        assertNull(androidCalendar.getEvent(id))
+        assertNull(androidCalendar.getEvent(idDirty0))
+        assertNull(androidCalendar.getEvent(idDirtyNull))
+    }
+
+    @Test
+    fun test_markNotDirty() {
+        val idDirty0 = androidCalendar.addEvent(Entity(contentValuesOf(
+            Events.CALENDAR_ID to androidCalendar.id,
+            Events._ID to 1,
+            Events.DTSTART to System.currentTimeMillis(),
+            Events.DTEND to System.currentTimeMillis(),
+            Events.TITLE to "Some Event 1",
+            Events.DIRTY to 0,
+            AndroidEvent2.COLUMN_FLAGS to 123
+        )))
+        val idDirtyNull = androidCalendar.addEvent(Entity(contentValuesOf(
+            Events.CALENDAR_ID to androidCalendar.id,
+            Events._ID to 2,
+            Events.DTSTART to System.currentTimeMillis(),
+            Events.DTEND to System.currentTimeMillis(),
+            Events.TITLE to "Some Event 2",
+            Events.DIRTY to null,
+            AndroidEvent2.COLUMN_FLAGS to 123
+        )))
+
+        val updated = calendar.markNotDirty(321)
+        assertEquals(2, updated)
+
+        assertEquals(321, androidCalendar.getEvent(idDirty0)?.flags)
+        assertEquals(321, androidCalendar.getEvent(idDirtyNull)?.flags)
```

Comment:
Same here. It's more logical because it tests different situations of `markNotDirty` (as the test is named), but I'd still make two different tests: one `markNotDirty_dirtyNull` and one `markNotDirty_dirtyZero`.